Ginger/Ginger/AutomatePageLib/NewAutomatePage.xaml.cs

```diff
@@ -1089,20 +1106,73 @@ private void xSaveBusinessFlowBtn_Click(object sender, RoutedEventArgs e)
                 }
             }
 
-            Reporter.ToStatus(eStatusMsgKey.SaveItem, null, mBusinessFlow.Name,
+            try
+            {
+                Reporter.ToStatus(eStatusMsgKey.SaveItem, null, mBusinessFlow.Name,
                                       GingerDicser.GetTermResValue(eTermResKey.BusinessFlow));
-            WorkSpace.Instance.SolutionRepository.SaveRepositoryItem(mBusinessFlow);
-            Reporter.HideStatusMessage();
+                SwapLoadingPrefixText("Saving", false);
+
+                await WorkSpace.Instance.SolutionRepository.SaveRepositoryItemAsync(mBusinessFlow);
+            }
+            finally
+            {
+                Reporter.HideStatusMessage();
+                SwapLoadingPrefixText("Saving", true);
+            }
         }
 
-        private void xUndoChangesBtn_Click(object sender, RoutedEventArgs e)
+        public string loadingText { get; set; } = "Loading [BusinessFlow]...";
+
+        void SwapLoadingPrefixText(string swappedText, bool IsReset)
+        {
+            if (string.IsNullOrEmpty(swappedText))
+            {
+                return;
+            }
+
+            if (IsReset)
+            {
+                loadingText = loadingText.Replace(swappedText, "Loading");
+                xItemsTabsSection.Visibility = Visibility.Visible;
+                xItemsLoadingPnl.Visibility = Visibility.Collapsed;
+            }
+            else
+            {
+                loadingText = loadingText.Replace("Loading", swappedText);
+                xItemsTabsSection.Visibility = Visibility.Collapsed;
+                xItemsLoadingPnl.Visibility = Visibility.Visible;
+            }
+
+            xLoadWindowText.Text = loadingText;
+            ToggleProcessButtons(IsReset);
+        }
+
+        private async void xUndoChangesBtn_Click(object sender, RoutedEventArgs e)
         {
             if (CheckIfExecutionIsInProgress()) return;
 
-            if (Ginger.General.UndoChangesInRepositoryItem(mBusinessFlow, true))
+            if (Reporter.ToUser(eUserMsgKey.AskIfToUndoItemChanges, mBusinessFlow.ItemName) == eUserMsgSelection.Yes)
             {
-                mActivitiesPage.ListView.UpdateGrouping();
-                mBusinessFlow.SaveBackup();
+                SwapLoadingPrefixText("Undoing", false);
+
+                try
+                {
+                    Reporter.ToStatus(eStatusMsgKey.StaticStatusProcess, null, string.Format("Undoing changes for '{0}'...", mBusinessFlow.ItemName));
+                    await Task.Run(() => mBusinessFlow.RestoreFromBackup(true, true));
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5936174)